### PR TITLE
feat(slice-machine): setup onboarding visibility ab test experiment

### PR DESCRIPTION
**Pull request by BohdanOne**
**Resolves**: [DT-2294](https://linear.app/prismic/issue/DT-2294/improve-onboardingcard-visibility)

### Description

This PR introduces basic set up for a new onboarding card visibility ab-test.

### Checklist

- [ ] If my changes require tests, I added them.
- [ ] If my changes affect backward compatibility, it has been discussed.
- [ ] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

**Review by levimykel — APPROVED** [review]
LGTM 🚀

File: packages/slice-machine/src/hooks/useOnboardingCardVisibilityExperiment.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useExperimentVariant } from "@/hooks/useExperimentVariant";
+
+type UseOnboardingCardVisibilityExperimentReturnType = {
+  eligible: boolean;
+  variant?: string;
+};
+
+export function useOnboardingCardVisibilityExperiment(): UseOnboardingCardVisibilityExperimentReturnType {
```
Comment (xrutayisire) [inline packages/slice-machine/src/hooks/useOnboardingCardVisibilityExperiment.ts]: ⚠️ #issue

For me, our `hooks` folder should be only used for generic shared hook if possible.
You could move this directly to the related feature folder, right? Like `src/features/onboarding`.

I see that `usePromptToCreateContentExperiment` could also be moved to `src/features/changes`.
And also that `useNestedGroupExperiment` could be moved to `src/features/builder`.

Could you move the 3 so it's a bit cleaner in hooks?

**Review by xrutayisire — CHANGES_REQUESTED** [review]
Small comment on where we put the hooks for experiment.

File: packages/slice-machine/src/hooks/useOnboardingCardVisibilityExperiment.ts (hunk 1 above)
Comment (BohdanOne) [reply packages/slice-machine/src/hooks/useOnboardingCardVisibilityExperiment.ts]: sure, will do!

**Review by jomifepe — APPROVED** [review]
🚀

File: packages/slice-machine/src/hooks/useOnboardingCardVisibilityExperiment.ts (hunk 1 above)
Comment (BohdanOne) [reply packages/slice-machine/src/hooks/useOnboardingCardVisibilityExperiment.ts]: done

**Review by xrutayisire — APPROVED** [review]
LGTM 🎉

**Pull request merged**